vertx-web/src/main/java/io/vertx/ext/web/handler/impl/SessionHandlerImpl.java

```diff
@@ -215,11 +223,13 @@ private void addStoreSessionHandler(RoutingContext context) {
   private void createNewSession(RoutingContext context) {
     Session session = sessionStore.createSession(sessionTimeout, minLength);
     context.setSession(session);
-    Cookie cookie = Cookie.cookie(sessionCookieName, session.id());
-    cookie.setPath("/");
-    cookie.setSecure(sessionCookieSecure);
-    cookie.setHttpOnly(sessionCookieHttpOnly);
-    // Don't set max age - it's a session cookie
+    Cookie cookie = Cookie.cookie(sessionCookieName, session.id())
+      .setPath("/")
+      .setSecure(sessionCookieSecure)
+      .setHttpOnly(sessionCookieHttpOnly);
+    // set max age if user requested it - else it's a session cookie
+    if(sessionCookieMaxAge > 0)
```

Comment:
Even though https://www.owasp.org/index.php/Session_Management_Cheat_Sheet#Expire_and_Max-Age_Attributes recommends to use non persistent cookies, they do not recommend to always avoid persistent cookies. For this reason I'd like to see a small paragraph about this on the manual:

https://github.com/vert-x3/vertx-web/blob/master/vertx-web/src/main/java/io/vertx/ext/web/package-info.java#L724